src/main/java/com/google/api/codegen/util/StaticResourcesHandler.java

```diff
@@ -36,7 +37,16 @@ public StaticResourcesHandler(Map<String, String> resourceFilesMap) {
     ImmutableMap.Builder<String, byte[]> resources = ImmutableMap.builder();
     for (Map.Entry<String, String> entry : resourceFilesMap.entrySet()) {
       String resourcePath = SnippetSetRunner.SNIPPET_RESOURCE_ROOT + '/' + entry.getKey();
-      byte[] resource = ByteStreams.toByteArray(cl.getResourceAsStream(resourcePath));
+      InputStream resourceStream = cl.getResourceAsStream(resourcePath);
+      // Hack necessary to allow jar files to be included in the fat jar
+      if (resourceStream == null && resourcePath.endsWith(".jar")) {
```

Comment:
Where do you do the renaming back to `.jar` that you mentioned above?

(I'm probably missing something since I don't have the full context of what's going in here.)

Reply:
`entry.getValue()` contains the file path, which will be `.../gradle-wrapper.jar`. The byte content of the .zjar file will end up in a .jar file.